lib/datastore.js

```diff
@@ -296,22 +306,96 @@ class Datastore extends EventEmitter {
           if (options.onload) options.onload(err)
           else throw err
         })
+    } else this.autoloadPromise = null
+    /**
+     * Interval if {@link Datastore#setAutocompactionInterval} was called.
+     * @private
+     * @type {null|number}
+     */
+    this.autocompactionIntervalId = null
+  }
+
+  /**
+   * Queue a compaction/rewrite of the datafile.
+   * It works by rewriting the database file, and compacts it since the cache always contains only the number of
+   * documents in the collection while the data file is append-only so it may grow larger.
+   *
+   * @async
+   */
+  compactDatafileAsync () {
+    return this.executor.pushAsync(() => this.persistence.persistCachedDatabaseAsync())
+  }
+
+  /**
+   * Callback version of {@link Datastore#compactDatafileAsync}.
+   * @param {NoParamCallback} [callback = () => {}]
+   * @see Datastore#compactDatafileAsync
+   */
+  compactDatafile (callback) {
+    const promise = this.compactDatafileAsync()
+    if (typeof callback === 'function') callbackify(() => promise)(callback)
+  }
+
+  /**
+   * Set automatic compaction every `interval` ms
+   * @param {Number} interval in milliseconds, with an enforced minimum of 5000 milliseconds
+   */
+  setAutocompactionInterval (interval) {
+    const minInterval = 5000
+    const realInterval = Math.max(interval || 0, minInterval)
+
+    this.stopAutocompaction()
+
+    this.autocompactionIntervalId = setInterval(() => {
+      this.compactDatafile()
+    }, realInterval)
+  }
+
+  /**
+   * Stop autocompaction (do nothing if automatic compaction was not running)
+   */
+  stopAutocompaction () {
+    if (this.autocompactionIntervalId) {
+      clearInterval(this.autocompactionIntervalId)
+      this.autocompactionIntervalId = null
     }
   }
 
   /**
-   * Load the database from the datafile, and trigger the execution of buffered commands if any.
-   * @param {function} callback
+   * Callback version of {@link Datastore#loadDatabaseAsync}.
+   * @param {NoParamCallback} [callback]
+   * @see Datastore#loadDatabaseAsync
    */
   loadDatabase (callback) {
-    this.executor.push({ this: this.persistence, fn: this.persistence.loadDatabase, arguments: [callback] }, true)
+    const promise = this.loadDatabaseAsync()
+    if (typeof callback === 'function') callbackify(() => promise)(callback)
   }
 
   /**
-   * Async version of {@link Datastore#loadDatabase}.
+   * Stops auto-compaction, finishes all queued operations, drops the database both in memory and in storage.
+   * **WARNING**: it is not recommended re-using an instance of NeDB if its database has been dropped, it is
+   * preferable to instantiate a new one.
+   * @async
+   * @return {Promise}
+   */
+  dropDatabaseAsync () {
+    return this.persistence.dropDatabaseAsync() // the executor is exceptionally used by Persistence
```

Comment:
C'est incohérent avec le compactDatafile que ici l'executor soit utilisé par persistence, alors que dans compactDatafile c'est par le datastore.

Patatable

Reply:
Je sais, il y a une raison : le dropDatabase fait usage interne de l'executor, il ne décore pas simplement une fonction de push, il fait un truc plus malin qui est fait de façon synchrone puis async.. ça demanderait de remonter la fonction entière côté datastore, why not 🤷  mais patate.